python/ray/serve/_private/deployment_state.py

```diff
@@ -2564,13 +2564,15 @@ def get_deployment_details(self, id: DeploymentID) -> Optional[DeploymentDetails
             )
 
     def get_deployment_statuses(
-        self, ids: List[DeploymentID] = None
+        self, ids: Optional[List[DeploymentID]] = None
     ) -> List[DeploymentStatusInfo]:
-        statuses = []
-        for id, state in self._deployment_states.items():
-            if not ids or id in ids:
-                statuses.append(state.curr_status_info)
-        return statuses
+        if ids is None:
+            return [
+                state.curr_status_info for state in self._deployment_states.values()
+            ]
+        else:
+            states = (self._deployment_states.get(id) for id in ids)
+            return [state.curr_status_info for state in states if state]
```

Comment:
Loop over the `ids` instead of over the deployments. This actually goes from `O(number of deployments * number of ids)` down to `O(number of ids)`, because the existing code was checking `id in ids` for each existing deployment, which is `O(number of ids)` since `ids` is a list.

Reply:
Is this changing the behavior ?

Reply:
Oh, nvm, it's the same.